src/50_Samples_%26_Templates/Star_Rating.html

```diff
@@ -0,0 +1,141 @@
+
+  <!--
+    #### CSS for the star rating
+  -->
+  <!--
+    This star rating implementation overlays "☆" Unicode characters precisely over radio buttons. Inline comments below describe the rules in detail.
+  -->
+  <style amp-custom>
+    .rating {
+      --star-size: 3em;  /* use CSS variables to calculate dependent dimensions later */
+      padding: 0;  /* to prevent flicker when mousing over padding */
+      border: none;  /* to prevent flicker when mousing over border */
+      unicode-bidi: bidi-override; direction: rtl;  /* for CSS-only style change on hover */
+      text-align: left;  /* revert the RTL direction */
+      user-select: none;  /* disable mouse/touch selection */
+      font-size: 3em;  /* fallback - IE doesn't support CSS variables */
+      font-size: var(--star-size);
+      cursor: pointer;
+      /* disable touch feedback on cursor: pointer - http://stackoverflow.com/q/25704650/1269037 */
+      -webkit-tap-highlight-color: rgba(0,0,0,0);
+      -webkit-tap-highlight-color: transparent;
+      /* border: 1px solid red;  /* debug */
+      margin-bottom: 1em;
+    }
+
+    /* the stars */
+    .rating > label {
+      display: inline-block;
+      position: relative;
+      width: 1.1em;  /* magic number to overlap the radio buttons on top of the stars */
+      width: calc(var(--star-size) / 3 * 1.1);
+      /* border: 1px solid green;  /* debug */
+    }
+    .rating > *:hover,
+    .rating > *:hover ~ label,
+    .rating:not(:hover) > input:checked ~ label {
+      color: transparent;  /* reveal the contour/white star from the HTML markup */
+      cursor: inherit;  /* avoid a cursor transition from arrow/pointer to text selection */
+    }
+    .rating > *:hover:before,
+    .rating > *:hover ~ label:before,
+    .rating:not(:hover) > input:checked ~ label:before {
+      content: "★";
+      position: absolute;
+      left: 0;
+      color: gold;
+    }
+
+    .rating > input {
+      position: relative;
+      transform: scale(3);  /* make the radio buttons big; they don't inherit font-size */
+      transform: scale(var(--star-size));
+      /* the magic numbers below correlate with the font-size */
+      top: -0.5em;  /* margin-top doesn't work */
+      top: calc(var(--star-size) / -6);
+      margin-left: -2.5em;  /* overlap the radio buttons exactly under the stars */
+      margin-left: calc(var(--star-size) / -6 * 5);
+
+      z-index: 2;  /* bring the button above the stars so it captures touches/clicks */
+      opacity: 0;  /* comment to see where the radio buttons are */
+    }
+    form.amp-form-submit-error [submit-error] {
+      color: red;
+    }
+  </style>
+</head>
+
+<body>
+
+  <h1>CSS-only keyboard-accessible rating widget</h1>
```

Comment:
This will never show up as it is hidden by default (no associated docs). However, it'd be good to enable the preview mode via metadata where this would be really helpful.